frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,275 @@ export class OutputArtifactLoader {
+    let res: GetExecutionsByContextResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getExecutionsByContext(request);
+    } catch (err) {
+      err.message = 'Failed to getExecutionsByContext: ' + err.message;
+      return err;
+    }
+
+    const executionList = (res && res.getExecutionsList()) || [];
+    const foundExecution = executionList.find(execution => {
+      const properties = execution.getPropertiesMap();
+      const executionPodName = properties.get('kfp_pod_name');
+      if (!executionPodName) {
+        return false;
+      }
+      const executionState = properties.get('state');
+      if (!executionState) {
+        return false;
+      }
+      return (
+        executionPodName.getStringValue() === argoPodName &&
+        executionState.getStringValue() === 'complete'
+      );
+    });
+    if (!foundExecution) {
+      logger.verbose("Couldn't find corresponding execution in context");
+      throw new Error("Couldn't find corresponding execution in context");
+    }
+    return foundExecution;
+  }
+
+  public static async getOutputArtifactsInExecution(execution: Execution): Promise<Artifact[]> {
+    const executionId = execution.getId();
+    if (!executionId) {
+      throw new Error('Execution must have an ID');
+    }
+
+    const request = new GetEventsByExecutionIDsRequest();
+    request.addExecutionIds(executionId);
+    let res: GetEventsByExecutionIDsResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getEventsByExecutionIDs(request);
+    } catch (err) {
+      err.message = 'Failed to getExecutionsByExecutionIDs: ' + err.message;
+      return err;
+    }
+
+    const eventsList = (res && res.getEventsList()) || [];
+    const outputArtifactIds: number[] = [];
+    eventsList.forEach(event => {
+      if (event.getType() === Event.Type.OUTPUT) {
+        const artifactId = event.getArtifactId();
+        if (artifactId) {
+          outputArtifactIds.push(artifactId);
+        }
+      }
+    });
+
+    const artifactsRequest = new GetArtifactsByIDRequest();
+    outputArtifactIds.forEach(artifactId => artifactsRequest.addArtifactIds(artifactId));
+    let artifactsRes: GetArtifactsByIDResponse;
+    try {
+      artifactsRes = await Api.getInstance().metadataStoreService.getArtifactsByID(
+        artifactsRequest,
+      );
+    } catch (artifactsErr) {
+      artifactsErr.message = 'Failed to getArtifactsByID: ' + artifactsErr.message;
+      return artifactsErr;
+    }
+
+    const artifactsList = (artifactsRes && artifactsRes.getArtifactsList()) || [];
+    return artifactsList;
+  }
+
+  public static async getArtifactTypes(): Promise<ArtifactType[]> {
+    const request = new GetArtifactTypesRequest();
+    let res: GetArtifactTypesResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getArtifactTypes(request);
+    } catch (err) {
+      err.message = 'Failed to getArtifactTypes: ' + err.message;
+      return err;
```

Comment:
```suggestion
      throw err;
```

Note, in typescript caught errors have `any` type. This is a trap that people may easily fall into. Therefore, even though you are returning an Error, which is not of type ArtifactType[]. The type checker doesn't emit any errors.